### PR TITLE
arch/armv[7|8]m: enhance the clang support

### PR DESCRIPTION
**Pull request by anchao**
## Summary

arch/armv[7|8]m: enhance the clang support

## Impact

arm clang

## Testing

ToolChain:
https://github.com/ARM-software/LLVM-embedded-toolchain-for-Arm

./tools/configure.sh lm3s6965-ek:qemu-flat
CONFIG_ARMV7M_TOOLCHAIN_CLANGL=y

### CODE REVIEW

**Comment by anchao** [comment]
Depends on: https://github.com/apache/incubator-nuttx-apps/pull/1140

File: arch/arm/src/armv8-m/Toolchain.defs [hunk 1]
```diff
@@ -161,26 +222,21 @@ ifneq ($(CONFIG_CXX_RTTI),y)
...
+EXTRA_LIBS += $(LIBGCC)
 
 ifneq ($(CONFIG_LIBM),y)
-  EXTRA_LIBS += ${wildcard ${shell $(CC) $(ARCHCPUFLAGS) --print-file-name=libm.a}}
+  EXTRA_LIBS += $(wildcard $(shell $(CC) $(ARCHCPUFLAGS) --print-file-name=libm.a))
 endif
 
 ifeq ($(CONFIG_LIBSUPCXX),y)
-  EXTRA_LIBS += ${wildcard ${shell $(CC) $(ARCHCPUFLAGS) --print-file-name=libsupc++.a}}
+  EXTRA_LIBS += $(wildcard $(shell $(CC) $(ARCHCPUFLAGS) --print-file-name=libsupc++.a))
```
Comment (xiaoxiang781216) [inline arch/arm/src/armv8-m/Toolchain.defs:241]: let's create a new patch to replace all ${/$} with $(/$).

Comment (anchao) [reply arch/arm/src/armv8-m/Toolchain.defs:241]: Done

**Review by xiaoxiang781216 — APPROVED** [review]

**Review by xiaoxiang781216 — APPROVED** [review]

**Review by pkarashchenko — APPROVED** [review]

**Comment by pkarashchenko** [comment]
14.0.0 release was started 20h ago. I will check if `--print-libgcc-file-name` bug is still there as soon as artifacts will be available for downloading,

**Comment by xiaoxiang781216** [comment]
> 14.0.0 release was started 20h ago. I will check if `--print-libgcc-file-name` bug is still there as soon as artifacts will be available for downloading,

where do you find 14.0.0 release?

**Comment by pkarashchenko** [comment]
> > 14.0.0 release was started 20h ago. I will check if `--print-libgcc-file-name` bug is still there as soon as artifacts will be available for downloading,
> 
> where do you find 14.0.0 release?

It is still not ready. Most probably the testing is in progress, but the release branch is ready, so I think the official release and artifacts will come in a week or two.

Screenshot 2022-04-15 at 12 08 34

**Comment by xiaoxiang781216** [comment]
Nice! I just open a tick to ask the new release(https://github.com/ARM-software/LLVM-embedded-toolchain-for-Arm/issues/108). It could be closed now:).

**Comment by pkarashchenko** [comment]
@xiaoxiang781216 Clang 14.0 is now available https://github.com/ARM-software/LLVM-embedded-toolchain-for-Arm/releases/tag/release-14.0.0

**Comment by xiaoxiang781216** [comment]
Yes, @anchao  is trying the new release.

**Pull request merged**